projects/ev3dev-jessie/rpi-base/root/boot/flash/config.txt

```diff
@@ -27,6 +27,10 @@
 # by default the Bluetooth uses the same UART as BrickPi
 #dtoverlay=pi3-disable-bt
 
+# uncomment only if you are using BrickPi (not BrickPi+) 
+# and want to use Input Port 5 for I2C/NXT sensor 
+#dtparam=i2c_baudrate=9600
```

Comment:
Yes, you are right in both. Probably it's better to run at higher if possible.
I only have CruizCore little to test here, it works ok at 9600 and higher, but starting from some baudrate it stops working properly (no errors in dmesg but -1 on angle value for example all the time)

Mindstorms NXT Hardware Developper Kit (page 9) states that: 
> I2C communication is running at 9600 bit/s

So at least in theory I2C/NXT sensors should at least work at this (slow!) speed and were tested for it.

But yes, I will update the docs and mention the somebody may want to tweak the value.